stream_alert/alert_processor/outputs/carbonblack.py

```diff
@@ -0,0 +1,108 @@
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+from collections import OrderedDict
+from cbapi.response import BannedHash, Binary, CbResponseAPI
+from stream_alert.alert_processor import LOGGER
+
+from stream_alert.alert_processor.outputs.output_base import (
+    OutputDispatcher,
+    OutputProperty,
+    StreamAlertOutput
+)
+
+
+@StreamAlertOutput
+class CarbonBlackOutput(OutputDispatcher):
+    """CarbonBlackOutput handles all alert dispatching for CarbonBlack"""
+    __service__ = 'carbonblack'
+
+    @classmethod
+    def get_user_defined_properties(cls):
+        """Get properties that must be asssigned by the user when configuring a new CarbonBlack
+        output.  This should be sensitive or unique information for this use-case that needs
+        to come from the user.
+
+        Returns:
+          OrderedDict: Contains various OutputProperty items
+        """
+        return OrderedDict([
+            ('descriptor',
+             OutputProperty(description='a short and unique descriptor for this'
+                                        ' carbonblack output')),
+            ('url',
+             OutputProperty(description='URL to the CB Response server [https://hostname]:',
+                            mask_input=True,
+                            cred_requirement=True)),
+            ('ssl_verify',
+             OutputProperty(description='Use SSL/TLS certificate validation [Y/N]:',
+                            mask_input=True,
+                            cred_requirement=True)),
+            ('token',
+             OutputProperty(description='API token (if unknown, leave blank):',
+                            mask_input=True,
+                            cred_requirement=True)),
+        ])
+
+    def dispatch(self, alert, descriptor):
+        """Send ban hash command to CarbonBlack
+
+        Args:
+            alert (Alert): Alert instance which triggered a rule
+            descriptor (str): Output descriptor
+
+        Returns:
+            bool: True if alert was sent successfully, False otherwise
+        """
+        if not alert.context:
+            LOGGER.error('[%s] Alert must contain context to run actions', self.__service__)
+            return self._log_status(False, descriptor)
+
+        creds = self._load_creds(descriptor)
+        if not creds:
+            return self._log_status(False, descriptor)
+
+        client = CbResponseAPI(**creds)
+
+        # Get md5 hash 'value' from streamalert's rule processor
+        action = alert.context.get('carbonblack', {}).get('action')
+        if action == 'ban':
```

Comment:
Should we extract the different actions into private instance methods? Do we see the list growing?

Reply:
this suggestion could also potentially make testing a bit easier

Reply:
Yeah, that's a good idea. I'm ok if it doesn't make it in this PR, but I think it's a pretty easy fix

Reply:
I do not see the list going in the near future.  Given other priorities we can slate this as an improvement in a future PR.